src/services/ERC20Service.ts

```diff
@@ -0,0 +1,103 @@
+import BN from 'bn.js';
+import { ethers } from 'ethers';
+import { Contract } from 'web3-eth-contract';
+import { toWei } from 'web3-utils';
+
+import Token from '@/models/Token';
+import { NetworkProvider } from '@/types/enums';
+import { deployLimitedSupplyERC20Contract, deployUnlimitedSupplyERC20Contract, getProvider } from '@/util/network';
+import TransactionService from './TransactionService';
+
+export const create = async (params: CreateERC20Params) => {
+    let response: { token: Contract; receipt: ethers.providers.TransactionReceipt };
+    const { admin } = getProvider(params.network);
+
+    if (Number(params.totalSupply) > 0) {
+        const { token, receipt } = await deployLimitedSupplyERC20Contract(
+            params.network,
+            params.name,
+            params.symbol,
+            admin.address,
+            toWei(new BN(params.totalSupply)),
+        );
+
+        response = { token, receipt } as any;
+    } else {
+        const { token, receipt } = await deployUnlimitedSupplyERC20Contract(
+            params.network,
+            params.name,
+            params.symbol,
+            admin.address,
+        );
+
+        response = { token, receipt } as any;
+    }
+
+    await Token.create({
+        name: params.name,
+        symbol: params.symbol,
+        network: params.network,
+        totalSupply: params.totalSupply,
```

Comment:
@GarfDev I assume this is stored for caching purposes? 

That would work for the LimitedSupply variant, but not for UnlimitedSupply tokens. For this variant the supply will grow with every minted token. I'd suggest to not store it in the db, but read it from the contract instead.